alf/utils/entropy_estimator.tex

```diff
@@ -0,0 +1,55 @@
+\documentclass{article}
+\usepackage{amsmath, amsthm, amssymb}
+\usepackage{mathbbol}
+
+\newcommand{\tr}{\mathrm{tr}}
+\newcommand{\erf}{\mathrm{erf}}
+\newcommand{\pian}[2]{\frac{\partial #1}{\partial #2}}
+\newcommand{\vect}[1]{\left[\begin{array}{c} #1 \end{array}\right]}
+\newcommand{\vectwo}[2]{\left[ \begin{array}{c c} #1 & #2 \\ \end{array} \right]}
+\newcommand{\innerprod}[2]{\left< #1 , #2 \right> }
+\newtheorem{theorem}{Theorem}
+\newtheorem{lemma}{Lemma}
+\newtheorem{corollary}{Corollary}
+	
+
+\title{Subtleties of estimating entropy}
+\author{Wei Xu}
+\begin{document}
+	\maketitle
+
+For some algorithms, we need to calculate the entropy and its derivative. If there is no analytic formula for the entropy, we can resort to sampling. Given the definition of entropy:
+\begin{equation}
+H(p) = E_{x\sim p_\theta}(-\log p_\theta(x))
+\end{equation}
+we can see that $-\log p(x)$ is an unbiased estimator of $H$ if $x$ is sampled from $p$. It is tempting to use $-\pian{\log p_\theta(x)}{\theta}$ as an estimator of $\pian{H}{\theta}$. However, it is wrong, as shown in the following: 
+\begin{eqnarray*}
+&&	E_{x\sim p_\theta}\left(\pian{\log p_\theta(x)}{\theta}\right) = \int \pian{\log p_\theta(x)}{\theta} p_\theta(x) dx \\
+&=& \int \pian{p_\theta(x)}{\theta} dx = \pian{}{\theta} \int p_\theta(x) dx = \pian{1}{\theta} = 0

```

Comment:
On a side note, this is exactly why we can't directly do SGD to optimize the RL object. Instead we have to do policy gradient or apply a reparameterization trick with SGD (only feasible in this case if the reward function is differentiable).